exporter/v1_compatibility.go

```diff
@@ -815,96 +819,162 @@ func myState(ctx context.Context, client *mongo.Client) (prometheus.Metric, erro
 	return metric, nil
 }
 
-func electionDate(m bson.M) prometheus.Metric {
-	replSetGetStatus, ok := m["replSetGetStatus"].(bson.M)
-	if !ok {
-		return nil
+func oplogStatus(ctx context.Context, client *mongo.Client) ([]prometheus.Metric, error) {
+	oplogRS := client.Database("local").Collection("oplog.rs")
+	type oplogRSResult struct {
+		Timestamp primitive.Timestamp `bson:"ts"`
 	}
-	members, ok := replSetGetStatus["members"].(primitive.A)
-	if !ok {
-		return nil
+	var head, tail oplogRSResult
+	headRes := oplogRS.FindOne(ctx, bson.M{}, options.FindOne().SetSort(bson.M{
+		"$natural": -1,
+	}))
+	if headRes.Err() != nil {
+		return nil, headRes.Err()
 	}
 
-	name := "mongodb_mongod_replset_member_election_date"
-	help := "The timestamp the node was elected as replica leader"
-
-	for _, member := range members {
-		if date, ok := member.(bson.M)["electionTime"].(primitive.Timestamp); ok {
-			labels := map[string]string{"name": member.(bson.M)["name"].(string)}
-			d := prometheus.NewDesc(name, help, nil, labels)
-			metric, _ := prometheus.NewConstMetric(d, prometheus.GaugeValue, float64(date.T))
+	if err := headRes.Decode(&head); err != nil {
+		return nil, err
+	}
+	tailRes := oplogRS.FindOne(ctx, bson.M{}, options.FindOne().SetSort(bson.M{
+		"$natural": 1,
+	}))
+	if tailRes.Err() != nil {
+		return nil, tailRes.Err()
+	}
 
-			return metric
-		}
+	if err := tailRes.Decode(&tail); err != nil {
+		return nil, err
 	}
 
-	return nil
+	headDesc := prometheus.NewDesc("mongodb_mongod_replset_oplog_head_timestamp",
+		"The timestamp of the newest change in the oplog", nil, nil)
+	headMetric := prometheus.MustNewConstMetric(headDesc, prometheus.GaugeValue, float64(head.Timestamp.T))
+
+	tailDesc := prometheus.NewDesc("mongodb_mongod_replset_oplog_tail_timestamp",
+		"The timestamp of the oldest change in the oplog", nil, nil)
+	tailMetric := prometheus.MustNewConstMetric(tailDesc, prometheus.GaugeValue, float64(tail.Timestamp.T))
+
+	return []prometheus.Metric{headMetric, tailMetric}, nil
+
 }
 
-func replicationLag(m bson.M) prometheus.Metric {
-	var primaryTS, selfTS primitive.Timestamp
-	var hasPrimary bool
-	var name, statestr string
+// replSetStatus keeps the data returned by the GetReplSetStatus command.
+type replSetStatus struct {
```

Comment:
We already have these structures in Toolkit: https://github.com/percona/percona-toolkit/blob/3.x/src/go/mongolib/proto/replstatus.go.
Can we use them instead?

Reply:
pingMS, lastHeartbeat and lastHeartbeatRecv is missing from there.

Reply:
Update them as well.

Reply:
Did PR there https://github.com/percona/percona-toolkit/pull/490

Reply:
Thanks. I've merged it. Could you update the dependency?